functions/index.js

```diff
@@ -44,6 +46,42 @@ exports.scheduledDatabaseCleanup = functions.pubsub.schedule('30 3 * * *').onRun
   });
 });
 
+exports.deleteUserData = functions.https.onCall((data, context) => {
+  const userId = data.userId;
```

Comment:
`data.userId` should be replaced with `context.auth.uid`, so that users can only delete their own accounts and are not able to delete the accounts of other users